hudi-hadoop-mr/src/main/java/org/apache/hudi/hadoop/HoodieFileGroupReaderBasedRecordReader.java

```diff
@@ -127,6 +131,14 @@ public HoodieFileGroupReaderBasedRecordReader(HiveReaderCreator readerCreator,
         props.setProperty(e.getKey(), e.getValue());
       }
     });
+    if (props.containsKey(REALTIME_SKIP_MERGE_PROP)
+        && !props.containsKey(MERGE_TYPE.key())) {
```

Comment:
Avoid using `.key()` for `containsKey` and `get`.  Use `ConfigUtils#containsConfigProperty` instead.